lib/game.rb

```diff
@@ -0,0 +1,43 @@
+class Game
+  def initialize
+    @rolls = [] 
+  end
+
+  def roll(pins)
+    @rolls.push(pins)
+  end
+
+  def points
+    frame_points = 0
+    rollIndex = 0
+    # calculates points per game frames(ten)
+    10.times do
+      if isStrike?(rollIndex)
+        frame_points +=  @rolls[rollIndex] + @rolls[rollIndex + 1] + @rolls[rollIndex + 2]
+        rollIndex += 1
+      elsif isSpare?(rollIndex)
+        frame_points += sparePoints(rollIndex)
+        rollIndex += 2
+      else
+        frame_points += @rolls[rollIndex] + @rolls[rollIndex + 1]
+        rollIndex += 2
+      end
+    end
+    frame_points
+  end
+
+def isSpare?(rollIndex)
```

Comment:
I think this is really clever. I didn't recognise that single values could be used ot calculate a spare / strike bonus, it's a great spot from you and well executed